### ISSUE TITLE
[US-5.2] AI Model Training and Fine-Tuning

### ISSUE THREAD

**Issue by MassPaol** [leonlolleonlol/DiscordPrototype#71]
As a developer, I want to train, fine-tune and save the AI model, in order to use it in auto-moderating tasks in the application.

Acceptance Criteria:
- [x] Training accuracy improves over time.
- [x] Fine-tuned model shows better accuracy or performance compared to the pre-trained model.
- [x] Saved model loads correctly when accessed.

This is a sub-issue of the epic user story #102
